File: Py/#8.py
# Time 15 minutes

problem = """
Write a function that takes an array of numbers (integers for the tests) and a target number. It should find two different items in the array that, when added together, give the target value. The indexes of these items should then be returned in a tuple / list (depending on your language) like so: (index1, index2).

For the purposes of this kata, some tests may have multiple answers; any valid solutions will be accepted.

The input will always be valid (numbers will be an array of length 2 or greater, and all of the items will be numbers; target will always be the sum of two different items from that array).

Based on: https://leetcode.com/problems/two-sum/

two_sum([1, 2, 3], 4) # returns (0, 2) or (2, 0)
two_sum([3, 2, 4], 6) # returns (1, 2) or (2, 1)
"""

def two_sum(numbers, target):
    res = list()
    for i, number in enumerate(numbers):
        for j, num in enumerate(numbers):
            if number + num == target and i is not j:
                res.append(i)
                res.append(j)
    return tuple(res[:2])

print(two_sum([11,11,29,1,4], 15))

File: Js/#8.js
// Time 20 minutes

const problem = `
We need to sum big numbers and we require your help.

Write a function that returns the sum of two numbers. The input numbers are strings and the function must return a string.

Example
add("123", "321"); -> "444"
add("11", "99");   -> "110"
Notes
The input numbers are big.
The input is a string of only digits
The numbers are positives
`;

function add(a, b) {
  let resultado = "";
  let acarreo = 0;

  let i = a.length - 1;
  let j = b.length - 1;

  while (i >= 0 || j >= 0 || acarreo > 0) {
    let digito1 = i >= 0 ? parseInt(a[i]) : 0;
    let digito2 = j >= 0 ? parseInt(b[j]) : 0;

    let suma = digito1 + digito2 + acarreo;

    resultado = (suma % 10) + resultado;
    acarreo = Math.floor(suma / 10);

    i--;
    j--;
  }

  return resultado;
}
